File: ido_sdk_flutter/ido_sdk/protocol_sdk/protocol_channel/ios/Classes/Model/IDOMenuListModel.swift
//
//  IDOMenuListModel.swift
//  protocol_channel
//
//  Created by hc on 2023/10/19.
//

import Foundation

@objcMembers
public class IDOMenuListModel: NSObject, IDOBaseModel {
    public var currentShowNum: Int
    public var items: [IDOMenuItem]?
    public var maxNum: Int
    public var maxShowNum: Int
    public var minShowNum: Int

    enum CodingKeys: String, CodingKey {
        case currentShowNum = "current_show_num"
        case items = "items"
        case maxNum = "max_num"
        case maxShowNum = "max_show_num"
        case minShowNum = "min_show_num"
    }

    public init(currentShowNum: Int, items: [IDOMenuItem]? = nil, maxNum: Int, maxShowNum: Int, minShowNum: Int) {
        self.currentShowNum = currentShowNum
        self.items = items
        self.maxNum = maxNum
        self.maxShowNum = maxShowNum
        self.minShowNum = minShowNum
    }
    
    public func toJsonString() -> String? {
        if let jsonData = try? JSONEncoder().encode(self) {
            let jsonString = String(data: jsonData, encoding: .utf8)
            return jsonString
        }
        return nil
    }
}

// MARK: - IDOMenuItem
@objcMembers
public class IDOMenuItem: NSObject, Codable {
    public var index: Int
    public var value: Int

    enum CodingKeys: String, CodingKey {
        case index = "index"
        case value = "value"
    }

    public init(index: Int, value: Int) {
        self.index = index
        self.value = value
    }
}

File: ido_sdk_flutter/ido_sdk/protocol_sdk/protocol_channel/android/src/main/kotlin/com/idosmart/model/IDOMenuListModel.kt
//
//  IDOMenuListModel.kt
//  protocol_channel
//
//  Created by hedongyang on 2023/10/30.
//

package com.idosmart.model

import com.google.gson.FieldNamingPolicy
import com.google.gson.GsonBuilder
import com.google.gson.annotations.SerializedName

///


open class IDOMenuListModel(
    currentShowNum: Int,
    items: List<IDOMenuItem>?,
    maxNum: Int,
    maxShowNum: Int,
    minShowNum: Int
) : IDOBaseModel {

    @SerializedName("current_show_num")
    var currentShowNum: Int = currentShowNum

    @SerializedName("items")
    var items: List<IDOMenuItem>? = items

    @SerializedName("max_num")
    var maxNum: Int = maxNum

    @SerializedName("max_show_num")
    var maxShowNum: Int = maxShowNum

    @SerializedName("min_show_num")
    var minShowNum: Int = minShowNum

    

    override fun toJsonString(): String {
        return GsonBuilder().create().toJson(this).toString()
    }

}

// MARK: - IDOMenuItem
open class IDOMenuItem(index: Int, value: Int) {
    @SerializedName("index")
    var index: Int = index

    @SerializedName("value")
    var value: Int = value

}